openedx_cas/backends.py

```diff
@@ -1,41 +1,133 @@
-"""Centralized Authentication Service backend based on the social-core library
-implementation.
+"""Centralized Authentication Service backend based on the social-core library implementation.
+
+Implements two backends, CASBackendOverride that allows us to perform the changes to the cas_validation
+method, making it compatible with the BaseAuth class.
 """
 from social_core.backends.base import BaseAuth
+from social_core.exceptions import AuthFailed
+from django.conf import settings
+from django_cas_ng.backends import CASBackend
+import urllib.parse
+from django_cas_ng.utils import get_cas_client
+from django.contrib.auth import get_user_model
+
+import logging
+
+UserModel = get_user_model()
 
+logger = logging.getLogger(__name__)
 
-class CASAuth(BaseAuth):
+class CASAuth(BaseAuth, CASBackend):
```

Comment:
@mariajgrimaldi I've refactored the old implementation (CASOverrideBackend) and mixed into CASAuth